src/control/cmd/dmg/pretty/telemetry.go

```diff
@@ -0,0 +1,178 @@
+//
+// (C) Copyright 2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package pretty
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/daos-stack/daos/src/control/lib/control"
+	"github.com/daos-stack/daos/src/control/lib/txtfmt"
+)
+
+// PrintMetricsListResp formats the MetricsListResp for human-readable output,
+// and writes it to the io.Writer.
+func PrintMetricsListResp(resp *control.MetricsListResp, out io.Writer) error {
+	if resp == nil {
+		return errors.New("nil response")
+	}
+
+	if len(resp.AvailableMetricSets) == 0 {
+		return nil
+	}
+
+	nameTitle := "Metric Set"
+	typeTitle := "Type"
+	descTitle := "Description"
+
+	tablePrint := txtfmt.NewTableFormatter(nameTitle, typeTitle, descTitle)
+	tablePrint.InitWriter(out)
+	table := []txtfmt.TableRow{}
+
+	for _, m := range resp.AvailableMetricSets {
+		row := txtfmt.TableRow{
+			nameTitle: m.Name,
+			typeTitle: m.Type.String(),
+			descTitle: m.Description,
+		}
+
+		table = append(table, row)
+	}
+
+	tablePrint.Format(table)
+
+	return nil
+}
+
+// PrintMetricsQueryResp formats a MetricsQueryResp for human-readable output,
+// and writes it to the io.Writer.
+func PrintMetricsQueryResp(resp *control.MetricsQueryResp, out io.Writer) error {
+	if resp == nil {
+		return errors.New("nil response")
+	}
+
+	if len(resp.MetricSets) == 0 {
+		return nil
+	}
+
+	for i, set := range resp.MetricSets {
+		if i > 0 { // a little space between metric sets
+			fmt.Fprintf(out, "\n")
+		}
+		fmt.Fprintf(out, "- Metric Set: %s (Type: %s)\n", set.Name, set.Type.String())
+
+		dw := txtfmt.NewIndentWriter(out)
+		fmt.Fprintf(dw, "%s\n", set.Description)
+
+		iw := txtfmt.NewIndentWriter(dw)
+		printMetrics(iw, set.Metrics, set.Type)
+
+	}
+	return nil
+}
+
+func printMetrics(out io.Writer, metrics []control.Metric, metricType control.MetricType) {
+	if len(metrics) == 0 {
+		fmt.Fprintf(out, "No metrics found\n")
+		return
+	}
+
+	nameTitle := "Metric"
+	labelTitle := "Labels"
+	valTitle := "Value"
+
+	tablePrint := txtfmt.NewTableFormatter(nameTitle, labelTitle, valTitle)
+	tablePrint.InitWriter(out)
+	table := []txtfmt.TableRow{}
+
+	for _, m := range metrics {
+		switch m.(type) {
```

Comment:
should be able to put the variable declaration here for reuse within the switch cases e.g. switch metric := m.(type) {

Reply:
Good catch!